lib/rubocop/cop/lint/object_equality_override.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Identifies `eql?` method definitions which do not
+      # check for argument class which might lead to runtime errors.
```

Comment:
There's a number of places in the PR where wording like this pops up but I don't think it's clear. Something like this might be better:

> Identifies `eql?` method definitions which do not check the argument's class is the same as, or a descendant of, the current class.

Since there are more "good" possibilities than are enumerated in `@example`, we should be more comprehensive.

Additionally "might lead to runtime errors" isn't a strong rationale IMO. What sort of runtime errors, and how does checking the class improve that?